libr/core/cmd_type.c

```diff
@@ -491,22 +491,28 @@ static void print_struct_union_in_c_format(Sdb *TDB, SdbForeachCallback filter,
 			if (var2) {
 				char *val = sdb_array_get (TDB, var2, 0, NULL);
 				if (val) {
+					char *arr = sdb_array_get (TDB, var2, 2, NULL);
 					if (multiline) {
 						r_cons_printf ("\t%s", val);
 						if (p && p[0] != '\0') {
 							r_cons_printf ("%s%s", strstr (val, " *")? "": " ", p);
+							if (atoi (arr)) {
+								r_cons_printf ("[%s]", arr);
+							}
 						}
+						r_cons_println (";");
 					} else {
-						r_cons_printf ("%s%s %s;", space, val, p);
+						r_cons_printf ("%s%s %s", space, val, p);
+						if (atoi (arr)) {
```

Comment:
what about using %d instead and show the actual parsed number instead of the string which can contain trailing non-numeric chars